### PR TITLE
[jvm-packages] change DeviceQuantileDmatrix into QuantileDMatrix

### PR DESCRIPTION
**Pull request by wbo4958**
To fix https://github.com/dmlc/xgboost/issues/8453

### CODE REVIEW

**Review by trivialfis — COMMENTED** [review]
Please mark the change as breaking.

The new `QuantileDMatrix` accepts both CPU and GPU based inputs, can we implement it on JVM side as well?

**Comment by wbo4958** [comment]
So you mean we also change the CPU way to build QuantileDMatrix instead of plain DMatrix?

**Comment by trivialfis** [comment]
Yes, similar to pyspark interface, we can have `QuantileDMatrix` as default for both `hist` and `gpu_hist`.

**Comment by wbo4958** [comment]
Ok, I will file another PR for it.

**Comment by wbo4958** [comment]
@trivialfis please help to check the CI why the xgboost-gpu tests run on windows and mac?

**Comment by trivialfis** [comment]
There's an undefined symbol: `XGQuantileDMatrixCreateFromCallbackImpl` when compiling without CUDA.

**Comment by wbo4958** [comment]
Thx, fixed it.

But I'm still curious why the test can pass in ubuntu? and I locally build CPU version with USE_CUDA=OFF, and it still pass

**Comment by trivialfis** [comment]
Thank you for the quick fix! triggered the CI.

**Comment by trivialfis** [comment]
Screenshot from 2022-11-22 14-59-21

@hcho3 Could you please help check whether the CI is working alright?

**Comment by hcho3** [comment]
@trivialfis An EC2 worker running Windows got stuck and hung for 10 hours. I had to manually kill it.

**Comment by wbo4958** [comment]
@trivialfis please help to trigger the CI. Thx

**Comment by trivialfis** [comment]
Done.

**Comment by wbo4958** [comment]
@trivialfis Seems the CI is not triggered.

**Comment by hcho3** [comment]
I triggered it again.

**Comment by wbo4958** [comment]
Does the python test have some issues?

```
E           ImportError: /opt/python/envs/cpu_test/lib/python3.8/site-packages/_cffi_backend.cpython-38-x86_64-linux-gnu.so: symbol ffi_type_uint32 version LIBFFI_BASE_7.0 not defined in file libffi.so.7 with link time reference
```

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/QuantileDMatrix.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    super(0);
+    long[] out = new long[1];
+    String conf = getConfig(missing, maxBin, nthread);
+    XGBoostJNI.checkCall(XGBoostJNI.XGQuantileDMatrixCreateFromCallback(
+        iter, (java.util.Iterator<ColumnBatch>)null, conf, out));
+    handle = out[0];
+  }
+
+  @Override
+  public void setLabel(Column column) throws XGBoostError {
```
Comment (trivialfis) [inline jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/QuantileDMatrix.java:32]: Could you please remind me what's the reason behind setters not being supported?

Comment (wbo4958) [reply jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/QuantileDMatrix.java:32]: this file is renamed by DeviceQuantileDMatrix which is not supporting setter before (when the PR was merged). Will file another PR to support setter if it indeed has supported that.

**Comment by wbo4958** [comment]
@trivialfis @hcho3 please help to trigger the CI. Thx

**Comment by hcho3** [comment]
Done

**Comment by wbo4958** [comment]
@hcho3 @trivialfis please help to trigger the CI again, I just merged the latest commits into this PR. Thx

**Comment by wbo4958** [comment]
@trivialfis  please help to check why python test failed in this CI, I suppose this PR won't touch any python code.

**Comment by hcho3** [comment]
@wbo4958 I re-created the machine env and re-triggered the tests. The CI is passing on the `master` branch, so hopefully it will remove the error.

**Comment by wbo4958** [comment]
Thx @hcho3, @trivialfis please help to review again. Thx

**Review by trivialfis — APPROVED** [review]

**Pull request merged**